bcda/auth/client/ssas.go

```diff
@@ -118,8 +118,29 @@ func (c *SSASClient) GetPublicKey(systemID int) ([]byte, error) {
 }
 
 // ResetCredentials PUTs to the SSAS /system/{systemID}/credentials endpoint to reset the system's credentials.
-func (c *SSASClient) ResetCredentials(systemID int) ([]byte, error) {
-	return nil, nil
+func (c *SSASClient) ResetCredentials(systemID string) ([]byte, error) {
+	req, err := http.NewRequest("PUT", fmt.Sprintf("%s/system/%s/credentials", c.baseURL, systemID), nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to delete credentials")
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to delete credentials")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 201 {
+		return nil, errors.New(fmt.Sprintf("failed to delete credentials. status code: %v", resp.StatusCode))
+	}
+
+	var respMap map[string]string
+	if err = json.NewDecoder(resp.Body).Decode(&respMap); err != nil {
+		return nil, errors.Wrap(err, "failed to delete credentials")
+	}
+
+	return []byte(respMap["client_secret"]), nil
```

Comment:
If this provider is only ever used for testing, only returning the secret is fine.  But if the value will somehow be given to a user/ACO, then there's value in giving the rest of the values returned by SSAS's `System.ResetSecret()`:
 - Client ID
 - Client Name
 - Expiration date for secret (this one is especially useful for a user)

I know the SSAS public router doesn't return more than the secret, but the implications are becoming more clear as it's being used.  😄